PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ab.cxx

```diff
@@ -112,129 +118,167 @@ struct MultiparticleCorrelationsAB // this name is used in lower-case format to
+    // *) Remaining event cuts which can be applied only after the loop over particles is performed:
+    if ((fSelectedTracks < ceh_a.fEventCuts[eSelectedTracks][eMin]) || (fSelectedTracks > ceh_a.fEventCuts[eSelectedTracks][eMax])) {
+      return;
+    }
+
+    // *) Calculate everything for selected events and particles:
+    CalculateEverything();
+
+    // *) Reset event-by-event quantities:
+    ResetEventByEventQuantities();
+
+  } // void processRecSim(...)
+  PROCESS_SWITCH(MultiparticleCorrelationsAB, processRecSim, "process both reconstructed and simulated information", true);
+
+  // -------------------------------------------
+
+  // C) Process only simulated data:
+  void processSim(aod::Collision const& collision, aod::BCs const&, TracksSim const& tracksSim, aod::McParticles const&) // called once per collision found in the time frame
+  {
+    // ...
+
+    // *) If I reached max number of events, ignore the remaining collisions:
+    if (!fProcessRemainingEvents) {
+      return; // TBI 20231008 Temporarily implemented this way. But what I really need here is a graceful exit from subsequent processing (which will also dump the output file, etc.)
+    }
+
+    // *) TBI 20231020 Temporary here (use configurable 'cfWhatToProcess' to set this flag corerectly):
+    if (!gProcessSim) {
+      LOGF(fatal, "in function \033[1;31m%s at line %d\033[0m", __PRETTY_FUNCTION__, __LINE__);
+    }
+
+    // *) Do all thingies before starting to process data (e.g. count number of events, fetch the run number, etc.):
+    Preprocess(collision, eSim);
+
+    // *) Fill event histograms for simulated data before event cuts:
+    FillEventHistograms(collision, tracksSim, eRec, eBefore);
+
+    // *) Event cuts:
+    if (!EventCuts(collision, tracksSim)) {
+      return;
+    }
+
+    // *) Fill event histograms for simulated data after event cuts:
+    FillEventHistograms(collision, tracksSim, eSim, eAfter);
+
+    // *) Main loop over reconstructed particles:
+    //    For simulated particles there is NO separate loop, if flag gProcessRecSim = true (via configurable "cfWhatToProcess"), MC info is accessed via label.
+    // MainLoopOverParticles(tracksSim, eSim); // TBI 20231020 it will NOT work this way + re-think the definition of tracksSim in the preamble
+
+    // *) Fill remaining event histograms for reconstructed and simulated data after event AND after particle cuts:
+    ceh_a.fEventHistograms[eSelectedTracks][eSim][eAfter]->Fill(fSelectedTracks);
+
+    // *) Remaining event cuts which can be applied only after the loop over particles is performed:
+    if ((fSelectedTracks < ceh_a.fEventCuts[eSelectedTracks][eMin]) || (fSelectedTracks > ceh_a.fEventCuts[eSelectedTracks][eMax])) {
+      return;
+    }
+
+    // *) Calculate everything for selected events and particles:
+    CalculateEverything();
+
+    // *) Reset event-by-event quantities:
     ResetEventByEventQuantities();
 
-  } // void process(...)
+  } // void processSim(...)
```

Comment:
In here you are replicating three times the same procedure. That's the role of the templating mechanism.
Use just one one templated function to do the processing and call it from the three PROCESS_SWITCH'ed functions which should only contain the call to that function. Clean and no code replicated. 
To use templates is not a fashion trend :-)

Reply:
That was in any case my intention, I left the comment, cite:

```TBI 20231017 There is a (minor) code bloat here obviously. Fix this when it becomes possible to post the last argument to PROCESS_SWITCH via variable (ideally Configurable)```

But since it's not possible to post the last argument to PROCESS_SWITCH via variable, I have introduced now templated Steer(...) function, which is the only one called in processRec(...), processRecSim(...) and processSim(...).